src/drf_yasg/inspectors/view.py

```diff
@@ -264,9 +264,12 @@ def get_response_schemas(self, response_serializers):
                 )
             else:
                 serializer = force_serializer_instance(serializer)
+                schema = self.serializer_to_schema(serializer)
+                if self.should_page():
```

Comment:
```suggestion
                if self.has_list_response():
                    schema = openapi.Schema(type=openapi.TYPE_ARRAY, items=schema)
                
                if self.should_page():
```

This needs another step to work with current code.